doc/dap.txt

```diff
@@ -1289,9 +1289,40 @@ An example:
 <
 
 To support async operations, the config providers functions are called
-within a coroutine and they can also return a `thread` which must be suspended
-and which will be resumed at most once and must then yield the configurations
-as list.
+within a coroutine.
+
+==============================================================================
+ON_CONFIG EXTENSIONS API                              *dap-providers-on_config*
+
+Plugins can pre-process the |dap-configuration| whenever a debug session
+starts.
+
+To do so, register a `on_config` hook in the `dap.providers.on_config` table.
+
+The key for the table is a `plugin-id`. Plugins should use their plugin name.
+Do _not_ use the `dap.` namespace. It is reserved for nvim-dap itself.
+
+The value for the table is a function that takes a |dap-configuration| as
+parameter and must return a |dap-configuration|.
+
+Before making modifications to the config you should mutate it, to ensure you
+don't make permanent changes to a configuration stored within
+`dap.configrations.<filetype>`.
+
+Example:
+>lua
+  local dap = require("dap")
+  dap.providers.on_config["dummy-noop"] = function(config)
```

Comment:
I'm still not sure about the overall naming. Other options:

- `dap.ext.on_config` and `dap.ext.configs`
- `dap.ext.on_config` and `dap.ext.config_providers`
- `dap.config_providers` and `dap.config_hooks`

Reply:
Moving it to `dap.listeners.on_config`: https://github.com/mfussenegger/nvim-dap/pull/1248